### ISSUE TITLE
Edit Log, line feeds disappear

### ISSUE THREAD

**Issue by geocachermgo** [cgeo/cgeo#15221]
### Describe your problem!

When I edit an existing log and go to edit log, the line feeds are removed from the log.

### How to reproduce?

Edit a log with empy line or/and line breaks.
Choose edit.

### Actual result after these steps?

line feeds are removed from the log text

### Expected result after these steps?

The base for the new log entry shall be the unchanged old log.

### Reproducible

Yes

### c:geo Version

2024.01.22-NB-b6c949b

### Additional Information

May other special characters are also altered?

**Comment by eddiemuc** [comment]
There is the need to alter the log for editing purposes. This is because we parse the log as HTML from gc.com, but it shall appear in "markup" language in the log edit field. There may be a problem in this altering process, although it should also handle Carriage Returns

I could not reproduce the problem with a log created on my own:

![image](https://github.com/cgeo/cgeo/assets/6909759/e581e390-7520-4a64-996d-646b12afee4b)
![image](https://github.com/cgeo/cgeo/assets/6909759/b083ec0b-27e3-40f7-b25a-af3df99430a9)

Perhaps we have another case here which I didn't consider.

@geocachermgo could you send me a pointer to an example log where this happens and which I can use to try to reproduce? If you don't want to share this here in public, you may contact me directly via gc.com (my cacher name is also EddieMuc).

**Comment by fm-sys** [comment]
Have you e.g. tested something like:
```
Testlog

---
Some text...
```

# without refreshing the cache, editing looks like:

![Screenshot_20240122_152701_cgeo geocaching](https://github.com/cgeo/cgeo/assets/64581222/f2217a5c-936a-4f71-bcfa-6950b67745e0)

# after refreshing the cache, editing looks like:

![Screenshot_20240122_152727_cgeo geocaching](https://github.com/cgeo/cgeo/assets/64581222/41bcc90d-c719-49c6-ad68-48b0b5595873)

**Comment by eddiemuc** [comment]
Who would have thought that edit log is so complex...

I tried to cover the most relevant formatting options (e.g. Bold, italic) but it is absolutely possible that I missed some. Strange, I thought I covered `---`

Ah, and the key of the bug is probably opening the log again BEFORE refresh. At this point the log is not html yet, but treated as auxh

**Comment by geocachermgo** [comment]
Yes, after reload the line breaks are not removed. But why should I reload, especially if I can edit in c:geo so the last changed has been done in cgeo.

**Comment by eddiemuc** [comment]
Fixed with PR #15223
